provider/cmd/app/app.go

```diff
@@ -497,22 +500,30 @@ func runMetricsServer(ctx context.Context, setupLog logr.Logger, opts HTTPServer
 		IdleTimeout:  opts.IdleTimeout,
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		<-ctx.Done()
-		setupLog.Info("Shutting down streaming server")
-		shutdownCtx, cancel := context.WithTimeout(context.Background(), time.Second)
+		setupLog.Info("Shutting down metrics server")
+		shutdownCtx, cancel := context.WithTimeout(context.Background(), opts.GracefulTimeout)
 		defer cancel()
 		locErr := srv.Shutdown(shutdownCtx)
 		if locErr != nil {
 			setupLog.Error(locErr, "metrics server wasn't shutdown properly")
+		} else {
+			setupLog.Info("Metrics server is shutdown")
 		}
-		setupLog.Info("Metrics server is shutdown")
 	}()
 
 	err := srv.ListenAndServe()
 	if err != nil && !errors.Is(err, http.ErrServerClosed) {
 		return fmt.Errorf("error listening / serving metrics server: %w", err)
 	}
 
+	setupLog.Info("Metrics server stopped serve new connections")
```

Comment:
Optional: This log statement appears unnecessary as it provides more of an informational message rather than being critical for the operation. Additionally, since we are already logging at line 508, it may not be very helpful in this context.

Reply:
I would like keep it here. it can be critical, because the shutdown process will stop accepting new connections, but the server can still live and serve existing connections, depending on the configuration graceful period. So you can have this weird state, when you cannot get metrics anymore but server is still running.